### ISSUE TITLE
locale support including _locale and time.tzname

### ISSUE THREAD

**Issue by youknowone** [RustPython/RustPython#3850]
- [x] `_locale` module #4558
- [ ] time.tzname

**Comment by bonjune** [comment]
#4054 

If we are talking about holding states such as `tzname` (which is initialized at module load time), this can be related.

**Comment by youknowone** [comment]
> The doc says tzname is determined at module load time. Does "module load time" mean those values to be initialized in the function below?

It may not be exactly, but you can regard it is.

**Comment by youknowone** [comment]
And yes. #4054 is generally related to module attributes and hidden data like Info types in sys module.

**Comment by JazzGlobal** [comment]
Noticed quite a few functions in [CPython Locale](https://github.com/python/cpython/blob/3.11/Modules/_localemodule.c) are not implemented in locale.rs. I don't have much experience with this type of work but would the general idea be to write a wrapper for say: [GetLocaleIInfoA](https://learn.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-getlocaleinfoa) since it is used throughout CPython's implementation of `getdefaultlocale` so it can be used in our implementation?

**Comment by DimitrisJim** [comment]
You are on Windows right? I haven't looked into how CPython does it (am out atm) but if this is calling some os provided function then yes, we typically wrap it and call into it after converting the args.

The function to be wrapped should be in windows.rs?

**Comment by JazzGlobal** [comment]
That's right, and gotcha. Looks like libc has bindings for `setlocale` but that's all im seeing

**Comment by arihant2math** [comment]
`time.tzname` is now supported on all platforms

**Comment by youknowone** [comment]
Great, both of the list done
